src/plugins/intel_cpu/src/nodes/stft.cpp

```diff
@@ -73,15 +77,95 @@ bool STFT::created() const {
+    const int64_t frame_size = (getSrcDataAtPortAs<const int32_t>(FRAME_SIZE_IDX))[0];
+    const int64_t frame_step = (getSrcDataAtPortAs<const int32_t>(FRAME_STEP_IDX))[0];
+
+    const auto is_signal_1D = signal_shape.size() == 1;
+    const size_t batch_size = is_signal_1D ? 1 : signal_shape[0];
+    const size_t signal_axis = is_signal_1D ? 0 : 1;
+    const auto signal_length = signal_shape[signal_axis];
+    const auto num_frames = static_cast<size_t>((signal_length - frame_size) / frame_step) + 1;
+    const auto frame_size_dim = static_cast<size_t>(frame_size);
+    const auto fft_out_shape = VectorDims{static_cast<size_t>((frame_size_dim / 2) + 1), 2};
+    const auto fft_out_shape_size = shape_size(fft_out_shape);
+
+    const auto window_length = window_shape[0] < frame_size_dim ? window_shape[0] : frame_size_dim;
+    std::vector<float> pad_window(frame_size, 0);
+    cpu_parallel_memcpy(pad_window.data() + (frame_size_dim - window_length) / 2,
+                        window,
+                        sizeof(float) * window_shape[0]);
+
+    float* dst = rdft_result;
+    const auto stft_shape = VectorDims{batch_size, num_frames, fft_out_shape[0], fft_out_shape[1]};
+    if (m_transpose_frames) {  // Store intermediate results
+        MemoryPtr dst_mem =
+            getScratchPadMem(std::make_shared<CpuBlockedMemoryDesc>(ov::element::f32, Shape{stft_shape}));
+        dst = dst_mem->getDataAs<float>();
+    }
+
+    parallel_for2d(batch_size, num_frames, [&](size_t batch, size_t frame_idx) {
+        size_t batch_in_start = batch * signal_length;
+        size_t batch_frames_out = batch * num_frames;
+
+        const auto frame_start = batch_in_start + frame_idx * frame_step;
+        const auto frame_end = frame_start + frame_size;
+        std::vector<float> signal_slice(signal + frame_start, signal + frame_end);
+        std::transform(signal_slice.begin(),
+                       signal_slice.end(),
+                       pad_window.begin(),
+                       signal_slice.begin(),
+                       std::multiplies<float>());
+
+        const auto result_idx = (batch_frames_out + frame_idx) * fft_out_shape_size;
+        auto twiddles = rdft_executor->generateTwiddles({static_cast<int>(signal_slice.size())}, fft_out_shape, {0});
+        rdft_executor->execute(signal_slice.data(),
+                               dst + result_idx,
+                               twiddles,
+                               1,
+                               {0},
+                               {static_cast<int>(frame_size)},
+                               {frame_size_dim},
+                               fft_out_shape,
+                               {1},
+                               {2, 1});
+    });
+    if (m_transpose_frames) {
+        const auto stft_transp_out_shape = VectorDims{batch_size, fft_out_shape[0], num_frames, fft_out_shape[1]};
+        transpose_out4d(reinterpret_cast<const uint8_t*>(dst),
```

Comment:
minor: the simpliest way to perform such things like transpose in CPU plugin is to use the following interface: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/cpu_memory.h#L190 with correctly initialized memory descriptors. In theory it should provides better performance.
Anyway I don't insist it should be refactored in bounds of this PR.

Reply:
Thank you for this hint! Worth to consider as a follow up, we have also briefly discussed potential further optimizations for the transpose here: https://github.com/openvinotoolkit/openvino/pull/26967#discussion_r1884016392